docker/scripts/functions.sh

```diff
@@ -154,14 +154,26 @@ function download_bios_configs() {
 	configure_image_cache_dns
 
 	echo "Downloading latest BIOS configurations"
-	curl --fail https://github-mirror.packet.net/downloads/bios-configs-latest.tar.gz --output bios-configs-latest.tar.gz
-	curl --fail https://github-mirror.packet.net/downloads/bios-configs-latest.tar.gz.sha256 --output bios-configs-latest.tar.gz.sha256
+	if ! curl --fail https://github-mirror.packet.net/downloads/bios-configs-latest.tar.gz --output bios-configs-latest.tar.gz; then
+		echo "Error: Download of BIOS configs tarball failed"
+		return 1
+	fi
+	if ! curl --fail https://github-mirror.packet.net/downloads/bios-configs-latest.tar.gz.sha256 --output bios-configs-latest.tar.gz.sha256; then
+		echo "Error: Download of BIOS configs checksum file failed"
+		return 1
+	fi
 
 	echo "Verifying BIOS configurations tarball"
-	sha256sum --check bios-configs-latest.tar.gz.sha256
+	if ! sha256sum --check bios-configs-latest.tar.gz.sha256; then
+		echo "Error: Verifying checksum of BIOS configs tarball failed"
+		return 1
+	fi
 
 	echo "Extracting BIOS configurations tarball"
-	tar -zxf bios-configs-latest.tar.gz
+	if ! tar -zxf bios-configs-latest.tar.gz; then
+		echo "Error: Extracting BIOS configs tarball failed"
```

Comment:
I'm not sure all these `if` and echos in the error case are really necessary. I feel like we already have enough info with the pre message that if there's an error that is caught by `set -e` we'd be able to figure things out. Otherwise why don't we do `if ! ...` for every line of code too?

Reply:
The point of this is to catch errors so we can report them as warnings and then skip the BIOS validation behavior, without triggering the autofail and failing the deprovision.

Reply:
sigh bash craziness bites me again... I now am aware that `set -e` is totally and globally ignored when part of complex pipelines even in sub function calls. So -e isn't actually on here since this is called in an if...